cli/build-public-library.js

```diff
@@ -0,0 +1,106 @@
+/*jshint node: true*/
+'use strict';
+
+const fs = require('fs-extra');
+const logger = require('winston');
+const rimraf = require('rimraf');
+const webpack = require('webpack');
+
+const stageTypeScriptFiles = require('./utils/stage-library-ts');
+const preparePackage = require('./utils/prepare-library-package');
+const webpackConfig = require('../config/webpack/build-public-library.webpack.config.js');
+const skyPagesConfigUtil = require('../config/sky-pages/sky-pages.config');
+
+function cleanTemp() {
+  rimraf.sync(skyPagesConfigUtil.spaPathTemp());
+}
+
+function cleanDist() {
+  rimraf.sync(skyPagesConfigUtil.spaPath('dist'));
+}
+
+function cleanAll() {
+  cleanTemp();
+  cleanDist();
+}
+
+function writeTSConfig() {
+  var config = {
+    'compilerOptions': {
+      'target': 'es5',
+      'module': 'es2015',
+      'moduleResolution': 'node',
+      'emitDecoratorMetadata': true,
+      'experimentalDecorators': true,
+      'allowSyntheticDefaultImports': true,
+      'sourceMap': true,
+      'noImplicitAny': true,
+      'declaration': true,
+      'skipLibCheck': true,
+      'lib': [
+        'dom',
+        'es6'
+      ],
+      'types': [
+        'jasmine',
+        'node'
+      ],
+      'outDir': skyPagesConfigUtil.spaPath('dist'),
+      'rootDir': skyPagesConfigUtil.spaPathTemp()
+    },
+    'files': [
+      skyPagesConfigUtil.spaPathTemp('index.ts')
+    ]
+  };
+
+  fs.writeJSONSync(skyPagesConfigUtil.spaPathTemp('tsconfig.json'), config);
+}
+
+function transpile(skyPagesConfig) {
+  const config = webpackConfig.getWebpackConfig(skyPagesConfig);
+  const compiler = webpack(config);
+
+  return new Promise((resolve, reject) => {
```

Comment:
I wonder if https://github.com/blackbaud/skyux-builder/blob/master/cli/build.js#L133-L163 is similar enough to factor this functionality out?